runtime/browser/web_application.h

```diff
@@ -40,7 +40,8 @@ class SplashScreen;
 class WebApplication : public WebView::EventListener {
  public:
   WebApplication(NativeWindow* window,
-                 common::ApplicationData* app_data);
```

Comment:
Do not remove 
constructor(WebApplication(NativeWindow* window, common::ApplicationData* app_data))
for SubClass of WebApplication

I recommand add a property to check a ownership of ewk_context_ 
bool has_ewk_context_


```
 WebApplication::WebApplication(
      NativeWindow* window,
      common::ApplicationData* app_data,
      Ewk_Context* context)
      : launched_(false),
        debug_mode_(false),
        verbose_mode_(false),
        ewk_context_(context),
        window_(window),
        appid_(app_data->app_id()),
        app_data_(app_data),
       locale_manager_(new common::LocaleManager()),
       terminator_(NULL),
       has_ewk_context_(false) {

}

 WebApplication::WebApplication(
      NativeWindow* window,
      common::ApplicationData* app_data) : WebApplication(window, app_data, ewk_context_new_with_injected_bundle_path(INJECTED_BUNDLE_PATH)), has_ewk_context_(true) {

}

```